src/config.py

```diff
@@ -0,0 +1,387 @@
+            raise ValueError('No relationship defined: {}. Check your .platform.app.yaml file.'.format(relationship))
+
+        if index not in range(len(self.relationshipsDef)):
+            raise ValueError('No index {} defined for relationship: {}.  '
+                             'Check your .platform.app.yaml file.'.format(index, relationship))
+
+        return self.relationshipsDef[relationship][index]
+
+    def variable(self, name, default=None):
+        """Returns a variable from the VARIABLES array.
+
+        Note: variables prefixed with `env`: can be accessed as normal environment variables.
+        This method will return such a variable by the name with the prefix still included.
+        Generally it's better to access those variables directly.
+
+        :param name: string
+            The name of the variable to retrieve.
+        :param default: mixed
+            The default value to return if the variable is not defined. Defaults to None.
+        :return: mixed
+            The value of the variable, or the specified default. This may be a string or an array.
+
+        """
+
+        if not self.is_valid_platform():
+
+            return default
+
+        return self.variablesDef[name] if name in self.variablesDef.keys() else default
+
+    def variables(self):
+        """Returns the full variables array.
+
+        If you're looking for a specific variable, the variable() method is a more robust option.
+        This method is for classes where you want to scan the whole variables list looking for a pattern.
+
+        :return: array
+            The full variables array
+        """
+
+        if not self.is_valid_platform():
+
+            RuntimeError('You are not running on Platform.sh, so the variables array is not available.')
+
+        return self.variablesDef
+
+    @property
+    def routes(self):
+        """Return the routes definition.
+
+        :return: array
+            The routes array, in PHP nested array form.
+        :exception RuntimeException:
+            If the routes are not accessible due to being in the wrong environment.
+        """
+        if not self.is_valid_platform():
+            raise RuntimeError('You are not running on Platform.sh, so routes are not available.')
+
+        if self.in_build():
+            raise RuntimeError('Routes are not available during the build phase.')
+
+        return self.routesDef
+
+    def get_route(self, route_id):
+        """Get route definition by route ID.
+
+        :param route_id: string
+            The ID of the route to load.
+        :return: array
+            The route definition. The generated URL of the route is added as a 'url' key.
+        :exception ValueError:
+            If there is no route by that ID, an exception is thrown.
+
+        """
+
+        for (url, route) in self.routes.items():
+
+            if route['id'] == route_id:
+
+                route['url'] = url
+
+                return route
```

Comment:
No need to space all of these out.  It's more readable without the blank lines.

Reply:
Same applies elsewhere in the file.  I'll stop calling them out to save review space. :smile: